modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/parser/hdbprocedure/XSKHDBProcedureParser.java

```diff
@@ -11,41 +11,100 @@
+import com.sap.xsk.hdb.ds.parser.XSKDataStructureParser;
+import com.sap.xsk.hdb.ds.synchronizer.XSKDataStructuresSynchronizer;
+import com.sap.xsk.utils.XSKCommonsConstants;
+import com.sap.xsk.utils.XSKCommonsUtils;
 import com.sap.xsk.utils.XSKHDBUtils;
+import custom.HanaProcedureListener;
+import models.HDBProcedureDefinitionModel;
+import models.HDBProcedureMissingPropertyException;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.antlr.v4.runtime.tree.ParseTreeWalker;
 import org.apache.commons.codec.digest.DigestUtils;
 import org.eclipse.dirigible.api.v3.security.UserFacade;
+import org.eclipse.dirigible.core.scheduler.api.ISynchronizerArtefactType;
 
-import com.sap.xsk.hdb.ds.api.IXSKDataStructureModel;
-import com.sap.xsk.hdb.ds.api.XSKDataStructuresException;
-import com.sap.xsk.hdb.ds.model.hdbprocedure.XSKDataStructureHDBProcedureModel;
-import com.sap.xsk.hdb.ds.parser.XSKDataStructureParser;
+import java.io.IOException;
+import java.sql.Timestamp;
+
+public class XSKHDBProcedureParser implements XSKDataStructureParser<XSKDataStructureHDBProcedureModel> {
+
+    private final XSKDataStructuresSynchronizer dataStructuresSynchronizer;
+    private final HDBProcedureSynchronizationArtefactType procedureSynchronizationArtefactType;
+    private final XSKHDBProcedureLogger procedureLogger;
+
+
+    public XSKHDBProcedureParser(XSKDataStructuresSynchronizer dataStructuresSynchronizer,
+                                 HDBProcedureSynchronizationArtefactType procedureSynchronizationArtefactType,
+                                 XSKHDBProcedureLogger procedureLogger) {
+        this.dataStructuresSynchronizer = dataStructuresSynchronizer;
+        this.procedureSynchronizationArtefactType = procedureSynchronizationArtefactType;
+        this.procedureLogger = procedureLogger;
+    }
+
+    @Override
+    public XSKDataStructureHDBProcedureModel parse(XSKDataStructureParametersModel parametersModel)
+            throws IOException, XSKDataStructuresException, XSKArtifactParserException {
+
+        String location = parametersModel.getLocation();
+
+        ParseTree parseTree = XSKHDBUtils.getParsedThree(parametersModel);
+
+        HanaProcedureListener listener = new HanaProcedureListener();
+
+        ParseTreeWalker parseTreeWalker = new ParseTreeWalker();
+        parseTreeWalker.walk(listener, parseTree);
+
+        HDBProcedureDefinitionModel antlr4Model = listener.getModel();
+        validateAntlrModel(antlr4Model, location);
+
+        return createModel(antlr4Model, parametersModel);
+    }
+
+    private XSKDataStructureHDBProcedureModel createModel(HDBProcedureDefinitionModel antlrModel,
+                                                          XSKDataStructureParametersModel params) {
+        XSKDataStructureHDBProcedureModel hdbProcedureModel = new XSKDataStructureHDBProcedureModel();
+        hdbProcedureModel.setSchema(antlrModel.getSchema());
```

Comment:
Isn't there an easy way to refactor `XSKDataStructureHDBProcedureModel` so that we pass everything in the constructor or create a builder for it? Using setters may lead to incorrectly initialized objects. Also, it's a better practice to have immutable objects (no setters) as it's often observed that setters may lead to spaghetti code when someone sets stuff into an object all over the code